devto-draft.md

```diff
@@ -1 +1,393 @@
+- updated() - Similar to React's componentDidUpdate. When you update data through your components, you want to send a POST request to your backend to keep things in sync
+- destroyed() - Similar to React's componentWillUnmount. You don't need to use this, but it helps clean up leftover event listeners.
+
+## Hooks
+
+Hooks makes React an extremely powerful framework. No longer did you have to use redux for handling state, you could just use `useState()` instead. No longer did you have to deal with `setState()`, there were a slew of other new improved methods to use.
+
+Vue has no equivalent to hooks, although there is an RFC at the time of writing this
+
+## Directives
+
+Vue's built-in directives make development a breeze. You could create iterable directives right inside your HTML.
+
+In React, this often meant making another component called "group(s)" and maybe one called "group".
+
+Vue, you can just dump the `v-for`, `v-if` etc right inside your HTML. Your logic coherently just makes sense looking at it.
+
+Alternatively, you can write your own directives! This lets you attach functionality to page if necessary, making it easier to develop faster
+
+One great use case for startups is to develop a reporting system. If a user clicks on a feature that doesn't yet exist, you can add a popup modal saying "Sorry we're working hard on this new feature! Come back later". In the background, you can make Axios request to notify users really do want that feature.
+
+## Mixins
+
+Some companies swear by these. It's a reusable code snippet library that can be implemented across multiple components. React out of the box does not have such a feature, you can alternatively use callback functions instead that are made globally available
+
+## Emitting Events
+
+Props are passed from parent to child component, via one-way data binding. To handle business logic at the parent level, from the child component, you generally emit events.
+
+With Vue, there are two different ways.
+
+1. You can pass a reference to the function from parent to child
+2. Emit an event from the child and capture the response in parent
+
+Most applications generally use #2. Example from a button that triggers a method in the parent component
+
+```vue
+<!-- Parent Component -->
+<template>
+  <ChildComponent @event-emitted="_handleUpdatedEvent" />
+</template>
+
+<script>
+import ChildComponent from "./components/ChildComponent.vue";
+
+export default {
+  components: {
+    ChildComponent
+  },
+  methods: {
+    _handleUpdatedEvent(val) {
+      console.log("Button was clicked");
+    }
+  }
+};
+</script>
+```
+
+```vue
+<!-- Child Component -->
+<template>
+  <div class="hello">
+    <button @click="$emit('event-emitted')">Click Me</button>
+  </div>
+</template>
+```
+
+## Global State Management
+
+Vue's CLI comes with the ability to add Vuex out of the box.
```

Comment:
Collect the sentences in a paragraph. I cannot really comment on the technical part of this as it's not my domain.